filclient/filc/main.go

```diff
@@ -380,26 +400,35 @@ var retrieveFileCmd = &cli.Command{
 			return err
 		}
 
-		stats, err := fc.RetrieveContent(ctx, miner, proposal)
+		stats, err := fc.RetrieveContent(cctx.Context, miner, proposal)
+		if err != nil {
+			return err
+		}
+
+		bserv := blockservice.New(node.Blockstore, offline.Exchange(node.Blockstore))
+		dserv := merkledag.NewDAGService(bserv)
+
+		dnode, err := dserv.Get(cctx.Context, c)
 		if err != nil {
 			return err
 		}
 
-		fmt.Println("retrieved content")
-		fmt.Println("Total Payment: ", stats.TotalPayment)
-		fmt.Println("Num Payments: ", stats.NumPayments)
-		fmt.Println("Size: ", stats.Size)
-		fmt.Println("Duration: ", stats.Duration)
-		fmt.Println("Average Speed: ", stats.AverageSpeed)
-		fmt.Println("Ask Price: ", stats.AskPrice)
-		fmt.Println("Peer: ", stats.Peer)
+		if err := os.WriteFile(outputStr, dnode.RawData(), 0755); err != nil {
```

Comment:
0755 is probably not what we want, 0644 is better (anything with the '1' bit set becomes executable)

Reply:
ah, okay. i was thinking you might download an executable, but it might be more secure to set it as non-executable and let the user change it if they want?